scripts/calculate_trending_scores_hourly.py

```diff
@@ -0,0 +1,142 @@
+import datetime
+from math import sqrt
+
+import _init_path  # noqa: F401 Imported for its side effect of setting PYTHONPATH
+
+from openlibrary.config import load_config
+from openlibrary.core import db
+from openlibrary.plugins.worksearch.code import execute_solr_query
+from openlibrary.plugins.worksearch.search import get_solr
+from scripts.solr_builder.solr_builder.fn_to_cli import FnToCLI
+
+# This script handles hourly updating of each works' z-score. The 'trending_score_hourly_23' field is
+# ignored, and the current count of bookshelves events in the last hour is the new trending_score_hourly[0]
+# value, with each other value incrementing up by one.
+
+# Trending_score_daily values are all fetched in order to calculate their mean and standard deviation.
+# The formula for the new z-score is [ z = sum(trending_score_hourly_*) - mean(trending_score_daily_*) /  (standard_deviation(trending_score_daily)] + 1).
+# The incrementation of the denominator by 1 serves several purposes:
+#       - It prevents divide-by-zero errors on objects with 0s in all trending_score_daily_* values, which is important,
+#         as they are instantiated to that upon the first time solr reindexes them with this new information.
+#       - It serves to deemphasize such values, as we're not particularly interested in a shift from
+#         0 events to 1 event.
+
+
+def fetch_works(current_hour: int):
+    ol_db = db.get_db()
+    query = "select work_id, Count(updated)"
+    query += "from bookshelves_books group by bookshelves_books.updated, work_id "
```

Comment:
I think this should be just group by `work_id` otherwise I think we might have count being 1!

```suggestion
    query += "from bookshelves_books group by work_id "
```

Reply:
We tested the query further, and this is the better iteration of it:

```sql
SELECT work_id, count(updated)
FROM bookshelves_books
WHERE updated >= localtimestamp - interval '1 hour'
GROUP BY work_id
```

Reply:
Excellent works, and thanks for the catch. Resolved!